api/app/models/fetch/predictions.py

```diff
@@ -192,6 +190,63 @@ async def _fetch_model_predictions_by_station_codes(model: ModelEnum, station_co
 
 
 async def fetch_model_predictions(model: ModelEnum, station_codes: List[int]):
-    """ Fetch 10 day global model weather predictions for a given station."""
+    """ Fetch model weather predictions for a given list of stations and a given model. """
     # Fetch predictions from the database.
     return await _fetch_model_predictions_by_station_codes(model, station_codes)
+
+
+async def _fetch_most_recent_historic_predictions_by_station_codes(model: ModelEnum, station_codes: List[int]) -> List[WeatherModelPrediction]:
+    """ Fetch the most recent historic model predictions from database based on each station's coordinates. """
+    # construct temporary helper dictionary of relevant weather stations
+    stations_dict = defaultdict(list)
+    stations = await app.stations.get_stations_by_codes(station_codes)
+    for station in stations:
+        stations_dict[station.code] = station
+
+    # construct helper dictionary of WeatherModelPredictions
+    # weather_model_predictions_dict is indexed by station_code, then by prediction_timestamp
+    weather_model_predictions_dict = defaultdict(lambda: defaultdict(list))
+
+    # We're only interested in the last 5 days
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+    five_days_ago = now - datetime.timedelta(days=5)
+    # send the query
+    session = app.db.database.get_session()
+    historic_predictions = app.db.crud.get_most_recent_historic_station_model_predictions(
+        session, station_codes, model, five_days_ago, now)
+    for prediction, prediction_model_run_timestamp, prediction_model in historic_predictions:
+        existing_prediction = weather_model_predictions_dict[prediction.station_code][
+            prediction.prediction_timestamp]
+        # insert the prediction into weather_model_predictions_dict if no entry for prediction.prediction_timestamp exists, or if
+        # prediction_model_run_timestamp.prediction_run_timestamp is more recent than the model_run time for the existing_prediction
+        if(existing_prediction == [] or existing_prediction.model_run.datetime < prediction_model_run_timestamp.prediction_run_timestamp):
```

Comment:
existing_prediction is an object, so it should check if it's None right?

Reply:
am now checking if existing_prediction is None